kiosk/src/Components/ErrorModal.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+import { tickEvent } from "../browserUtils";
+import "../Kiosk.css";
+
+
+interface IProps {
+    errorType: string;
+    errorDescription: string;
+    showing: boolean;
+    setShowing: (p: string) => void;
+}
+const ErrorModal: React.FC<IProps> = ({ errorType, errorDescription, showing, setShowing }) => {
+    const cancelClicked = () => {
+        const modal = document.getElementsByClassName("common-modal-container")[0];
+        while (modal.firstChild) {
+            modal.removeChild(modal.firstChild);
+        }
+        tickEvent("kiosk.scanError.dismissed");
+        setShowing("");
+    }
+
+    if (!showing) {
+        return null;
+    }
+
+    return (
```

Comment:
Can the rendering here be conditioned on `showing`? Rather than the DOM node removals above, let React handle removing the elements.

Reply:
Just an idea: It might be simpler to remove `showing`/`setShowing`, only render if `errorType` is set, and have `cancelClicked` clear its value.